server/container_update_resources.go

```diff
@@ -51,3 +56,57 @@ func toOCIResources(r *pb.LinuxContainerResources) *rspec.LinuxResources {
 		// TODO(runcom): OOMScoreAdj is missing
 	}
 }
+
+func (s *Server) updateContainerResources(ctx context.Context, c *oci.Container, resources *rspec.LinuxResources) (retErr error) {
+	if err := updateContainerSpec(c, resources); err != nil {
+		return err
+	}
+
+	defer func() {
```

Comment:
instead of cleaning up, I think we should have the update be atomic. We can write the file to some random location, then move it to the original spec location after the update succeeds